Update push plugin with NativeScript plugin seed guidelines

lini opened the pull request:
- Update version to 0.2.0 (major)
- change Android callback method signature (remove the first message parameter as it is unused)
- add demo app (TypeScript)
- move plugin source code to src/ directory
- make plugin source code TypeScript
- add publish/pack scripts
- add Travis CI integration
- merge #129 into the code
- update native Android library dependencies to fix problem reported in Android Oreo (8.0)

lini committed:
Add Vanilla TS demo app

lini committed:
Update readme with required Android configuration

lini committed:
Add Angular demo app

lini committed:
reference source code for plugin

lini committed:
Update definition for ios callback

It returns a JSON object, not a string

lini committed:
Add tslint

part of verification process

lini committed:
Added tslint script

verify requirement

lini committed:
Update license

Verify requirement

lini committed:
Update README

Verify requirements

lini committed:
update README

lini committed:
update README

final hopefully :smile:

lini committed:
add info in package.json

lini committed:
Add support for typescript

tsc build and preparation for ts conversion

lini committed:
update to plugin seed specs

- Move plugin source to src/
- remove angular demo (non-UI plugin, so no need for two demo apps)
- migrate plugint to TS
- add to vendor.js in demo app

lini committed:
update ios script

fix errors in demo app

lini committed:
add publish & pack scripts from seed

due to current problems with symlinks and nativescript-hook, demo app updated to reference plugin package instead of plugin source directory. pack.sh from the publish folder should be called before running demo app.

lini committed:
Update native Android library

apply https://github.com/NativeScript/push-plugin/pull/129

lini committed:
Update native android code.

- fix background notifications in Oreo.
- remove unneeded resources from native android output.

lini committed:
Add Ttravis CI configuration

lini committed:
fix publish scripts permissions

lini committed:
updated platform verisions in travis config

lini committed:
Update native Android project dependencies

Fix issue when returned token is null on first start of app and registering with Firebase

lini committed:
update src android gradle

- use supportVersion project property for appcompat library

- update demo package.json to have tslint dev dependency (needed for CI)

lini committed:
update Travis - add test google-services.json content

lini committed:
fix yml file syntax

lini committed:
update travis scripts

lini committed:
testing travis config

DimitarTachev commented on demo/app/bundle-config.ts:
```diff
@@ -0,0 +1,7 @@
+    require("bundle-entry-points");
+
+    // register application modules
+    global.registerModule("main-page", () => require("./main-page"));
```
require the plugin if we wanna test it with webpack.

DimitarTachev commented on demo/app/main-view-model.ts:
```diff
@@ -0,0 +1,93 @@
+import { Observable } from "tns-core-modules/data/observable";
+import * as pushPlugin from "nativescript-push-notifications";
+
+export class HelloWorldModel extends Observable {
```
HelloWorldModel?

DimitarTachev commented on demo/tsconfig.json:
```diff
@@ -0,0 +1,26 @@
+        "baseUrl": ".",
+        "paths": {
+            "*": [
+                "./node_modules/tns-core-modules/*",
```
We do not recommend using a tns-core-modules path.

DimitarTachev commented on publish/pack.sh:
```diff
@@ -0,0 +1,47 @@
+    npm i
+}
+
+pack() {
```
don't we need some native builds here?

DimitarTachev commented on src/package.json:
```diff
@@ -0,0 +1,51 @@
+    "url": "https://github.com/NativeScript/push-plugin.git"
+  },
+  "license": "Apache-2.0",
+  "homepage": "https://github.com/NativeScript/nativescript-imagepicker",
```
? :)

DimitarTachev commented on src/platforms/android/include.gradle:
```diff
@@ -11,8 +11,16 @@ android {
+    def firebaseMessagingVersion = project.hasProperty('firebaseMessagingVersion') ? project.firebaseMessagingVersion : '11.4.2'
+    def supportVer = "25.1.0";
 
+    if(project.hasProperty("supportVersion")) {
```
unify the project properties usage (ternary operator OR an if statement)

DimitarTachev commented on src/platforms/android/include.gradle:
```diff
@@ -11,8 +11,16 @@ android {
+       supportVer = supportVersion
+    }
+
+    println 'Push Plugin is using ' + supportVer + ' of the Android support library'
```
print both versions or none of them

lini committed:
Fixes

- remove print from include.gradle for Android
- fix homepage in package.json
- rename view model for demo page
- register plugin in bundle-config
- update travis labels

lini committed:
remove tns modules path from tsconfig

lini committed:
update pack script to explain about native library builds
